bindings/go/scip/internal/shared/shared.go

```diff
@@ -0,0 +1,16 @@
+// Package shared has functions used both by the old parser and the new parser.
+package shared
+
+func IsSimpleIdentifierCharacter(c rune) bool {
+	return c == '_' || c == '+' || c == '-' || c == '$' || ('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z') || ('0' <= c && c <= '9')
```

Comment:
Changing the ordering of the comparisons doesn't seem to have any noticeable changes in benchmarks, so leaving this code as-is to match the order in `scip.proto`.